modules/storage-rocksdb/src/main/java/org/apache/ignite/internal/storage/rocksdb/RocksDbMvPartitionStorage.java

```diff
@@ -853,24 +859,22 @@ protected ReadResult decodeEntry(byte[] key, byte[] value) {
 
                 @Override
                 public boolean hasNext() {
-                    return busy(() -> {
-                        throwExceptionIfStorageInProgressOfRebalance(state.get(), RocksDbMvPartitionStorage.this::createStorageInfo);
+                    assert rowIsLocked(rowId);
 
-                        return super.hasNext();
-                    });
+                    return super.hasNext();
                 }
 
                 @Override
                 public ReadResult next() {
-                    return busy(() -> {
-                        throwExceptionIfStorageInProgressOfRebalance(state.get(), RocksDbMvPartitionStorage.this::createStorageInfo);
+                    assert rowIsLocked(rowId);
 
-                        return super.next();
-                    });
+                    return super.next();
                 }
 
                 @Override
                 public void close() {
+                    assert rowIsLocked(rowId);
```

Comment:
We don't, that's me being paranoid. Should I remove the assertion?

Reply:
I don't have a hard opinion on this, but sometimes an (unrelated) error happening while trying to release a resource might be a problem (like masking a real error)

Reply:
this assertion should never happen in practice

Reply:
This assertion is for commenting purposes and for other developers to catch problems if they use cursors in unintended ways.
This particular assertion should never happen in production environment, because it only depends on code structure, there's no data dependency. Thus we can be sure about it